htdocs/product/stock/massstockmove.php

```diff
@@ -329,28 +329,46 @@
 					continue;
 				}
 				//var_dump($data);
-
+				$formproduct = new FormProduct($db);
```

Comment:
Each new allocate more memory. If you load 10000 line, you may have an out of memory.
Because we only need the instance to make a fetch (that erase old data), i think you can move the 4 new Xxx() here just before the while of the loop.